cmd/podman/images/scp.go

```diff
@@ -0,0 +1,410 @@
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to create new ssh session on %q", uri.Host)
+	}
+	defer session.Close()
+	if err != nil {
+		return "", err
+	}
+
+	if LoadSave { // if we are loading to the remote/ssh client
+		f, _ := os.Open(imageFileIn) // open image file to copy
+		defer f.Close()
+		n, err := scpD.CopyTo(dial, imageFileIn, imageFileOut)
+		if err != nil {
+			fmt.Println(n, "Bytes copied before error. Error while copying file ", err)
+		}
+
+		podman := "podman"
+		if v, found := os.LookupEnv("PODMAN_BINARY"); found {
+			podman = v // get path to podman bin
+		}
+		run := podman + " image load --input=" + imageFileOut // run ssh image load of the file copied via scp
+		var buffer bytes.Buffer
+		session.Stdout = &buffer
+		if err := session.Run(run); err != nil { // run the command declared above
+			return "", err
+		}
+		return buffer.String(), nil // return the output of image load on the machine, should be "Loaded Image(s)..."
+	} // else we are loading from the ssh client
+	podman := "podman"
+	if v, found := os.LookupEnv("PODMAN_BINARY"); found {
+		podman = v // get path to podman bin
+	}
+	run := podman + " image save " + imageFileOut + " --format=oci-archive --output=" + imageFileIn // run ssh image load of the file copied via scp. Files are reverse in thie case...
+	var buffer bytes.Buffer
+	session.Stdout = &buffer
+
+	if err := session.Run(run); err != nil { // run the command declared above
+		return "", err
+	}
+	n, err := scpD.CopyFrom(dial, imageFileIn, imageFileIn)
+	if err != nil {
+		fmt.Print(n, " bytes copied before err")
+		return "", err
+	}
+	return "", nil
+}
+
+// getUserInfo takes a ssh url and extracts user information
+func getUserInfo(uri *urlP.URL) (*urlP.Userinfo, error) {
+	var (
+		usr *user.User
+		err error
+	)
+	if u, found := os.LookupEnv("_CONTAINERS_ROOTLESS_UID"); found {
+		usr, err = user.LookupId(u)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to lookup rootless user")
+		}
+	} else {
+		usr, err = user.Current()
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to obtain current user")
+		}
+	}
+
+	pw, set := uri.User.Password() // check for a password set
+	if set {
+		return urlP.UserPassword(usr.Username, pw), nil
+	}
+	return urlP.User(usr.Username), nil
+}
+
+func parseLocation(args []string, cfg *config.Config) ([]string, []string, map[string]config.Destination) {
+	iden := []string{}
+	url := []string{}
+	serv := map[string]config.Destination{}
+	if strings.Contains(args[0], "::") || len(args) == 2 { // if we have a connection name specified
+		if len(args) == 2 && !(strings.Contains(args[1], "::") && strings.Contains(args[0], "::")) { // if an image is specified, this mean we are loading from our client
+			scpOpts.Connections = append(scpOpts.Connections, args[1])
+			scpOpts.ToRemote = true
+		} else if len(args) == 2 && (strings.Contains(args[1], "::") && strings.Contains(args[0], "::")) { // both remote clients
```

Comment:
(Personal nit: I’d find this a bit easier to follow if order of the checks were `args[0], args[1]`, not `args[1], args[0]`.)